### ISSUE TITLE
Problem with RaspPi kalite not loading/not opening ports

### ISSUE THREAD

**Issue by aetiologicCanada** [learningequality/ka-lite#4788]
I am having two problems with the kalite daemon installed with RaspPI_EN64_3.7. Static pages are fine, but KA pages are having two problems:

1) The rachel homepage uses possibly malformed URLS,e.g.:http://192.168.0.102/:8008/learn/khan/math/trigonometry/ 
I think the ...102/:8008 is incorrect and that it should be 102:8008. However, correctling this does not solve the problem 2
2) The daemon appears to load happily but  once loaded, I can't find the process and the port is closed. 

Here is a console dump:

pi@rachel ~ $ service kalite stop
/etc/init.d/kalite: 1: /etc/init.d/kalite: You: not found
Shutting down ka-lite!
Password:
INFO:kalite:RPi package selected.
kalite stopped pi@rachel ~
$ service kalite start
/etc/init.d/kalite: 1: /etc/init.d/kalite: You: not found Starting ka-lite!
Password:
INFO:kalite:RPi package selected.
INFO:kalite:Invalidating the web cache.
INFO:kalite:Running videoscan.
INFO:kalite:Preloading exercise data for language en.
INFO:kalite:Preloading content data for language en.
INFO:kalite:Preloading topic tree data for language en. 0 static files copied, 2396 unmodified. To access KA Lite from another connected computer, try the following address(es):
http://127.0.0.1:8008/
http://192.168.0.102:8008/

Suggestions welcome.

**Comment by 3kay** [comment]
Try removing the files in /home/pi/, kalite and rebooting.

**Comment by 3kay** [comment]
Sorry it is .kalite not coma as it appears on my mobile screen.

**Comment by MCGallaspy** [comment]
@aetiologicCanada thanks for the detailed report! What version of KA Lite are you using?

Can you also elaborate on this:
>2) The daemon appears to load happily but once loaded, I can't find the process and the port is closed.

What behavior do you expect? Are you looking for a "kalite" process, but you're not finding it? Is the app unavailable when you navigate to one of the suggested addresses (e.g. http://127.0.0.1:8008/)?

**Comment by benjaoming** [comment]
@aetiologicCanada I think there's something up with the init script from Rachel. Could you paste the contents of `/etc/init.d/kalite` here?

Our original version is called `/etc/init.d/ka-lite` and I think yours has been customized by Rachel so it's a little hard telling what the problem is. For instance, it's complaining about a command "You" not being there, sounds like a typo in the script.

As for the URLs that are wrong, this is most likely caused by a wrong setting somewhere, but exactly where are the URLs showing up wrongly? Could you post a screenshot of the page with wrong URLs and an indicator of which that are wrong?

**Comment by benjaoming** [comment]
Closing this due to lack of feedback.

@j-schwartz do you recognize any of this feedback as an issue in Rachel or KA Lite?

**Comment by j-schwartz** [comment]
Hi Guys - It probably had to do with this:https://racheloffline.org/pages/forum#!/rachel-pi:rachelpi64-en-khan-academy

I didn't totally understand it, but older RPI hardware was running KA-lite on port 7007, and the links wer e pointed to port 8008

**Comment by benjaoming** [comment]
Hi @j-schwartz 

It's supposed to run on port 7007 with nginx proxying on port 8008 :)

**Comment by j-schwartz** [comment]
Ah - in layman's terms - do I want people on the RPI using KA-lite via port 7007 or port 8008?

**Comment by benjaoming** [comment]
Supposing you're using the ka-lite-raspberry-pi .deb package, you definitely want to point to 8008, the reason that nginx is proxying is that it handles all static files, like videos, much more efficiently than the web server that KA Lite ships with... we needed this optimization because of the RPI platform's low specs.

**Comment by jamalex** [comment]
As long as Nginx is up and running, you'll want to have people using port 8008, as it will be more performant (static resources served by Nginx rather than Python).

**Comment by j-schwartz** [comment]
perfect, thanks guys.
